src/charm.py

```diff
@@ -37,13 +38,11 @@ def __init__(self, *args):
             )
         )
 
-        # TODO before ckf 1.9: Change this from SDI to the service-info lib
-        #  https://github.com/canonical/envoy-operator/issues/76
         self.grpc = self.charm_reconciler.add(
-            component=SdiRelationDataReceiverComponent(
+            component=K8sServiceInfoComponent(
                 charm=self,
-                name="relation:grpc",
-                relation_name="grpc",
+                relation_name=GRPC_RELATION_NAME,
+                refresh_event=self.on[GRPC_RELATION_NAME].relation_changed,
```

Comment:
See [this](https://github.com/canonical/kfp-operators/pull/436#discussion_r1569224916)